content/projects/oop-for-data-science/_index.md

```diff
@@ -1,3 +1,35 @@
 ---
 title: OOP for data science
+ready: False
 ---
+
+## Prerequisite
+
+You need to have completed the Multivariate Regression assignment using the salary dataset before doing this assignment ({{% contentlink "projects/data-science-specific/multivariate-linear-regression" %}}). You will use the data and the model you built in that assignment this OOP assignment.
+
+## Assignment
+It is efficient to put machine learning models and other data science techniques into classes so that we can reuse them later and change attributes without changing the code behind these models. Independent concepts can also be put into independent classes: for example, the functioning of a cross-validate class should not affect the functioning of a linear regression class.
+
+1. Create a metrics class that has methods to compute the Mean Squared Error, R-squared and Adjusted R-squared. The class should have the following attributes:
+
+- `x`: 1D or 2D array of features/predictor variables
+- `y`: 1D array containing the outcome variable
+- `model type` (e.g. linear regression)
+
+2. Create a diagnostic_plots class that has methods to plot the (1) residuals versus predicted values and (2) predicted versus observed values.
```

Comment:
please follow Python naming conventions, diagnostic_plots is not a suitable name for a class

material: https://pep8.org/#class-names